src/EventTeachingService.cpp

```diff
@@ -399,37 +393,31 @@ Processed EventTeachingService::handleLearnEvent(VlcbMessage *msg, unsigned int
     controller->sendGRSP(OPC_EVLRN, getServiceID(), CMDERR_INV_EV_IDX);
     return PROCESSED;
   }
-
-  // Is this an existing event? 
+// Is this a produced event that we know about?
+      // Search the events table by evindex = 1 for a value match with evval.
+  if ((evindex == 1) && (evval > 0))
+      {
+        byte index = module_config->findExistingEventByEv(evindex, evval);
+        if (index < module_config->EE_MAX_EVENTS)
+        {
+          module_config->writeEvent(index, &msg->data[1]);
+          // no need to write eventEV as, by definition, it hasn't changed
+          // recreate event hash table entry
+          module_config->updateEvHashEntry(index);
+          return PROCESSED;
+        }
+      }     
+      
+      // search for this NN, EN as we may just be adding an EV to an existing learned event 
```

Comment:
Please fix indentation here.

Reply:
OK

Reply:
Indentation is still out for lines 396-427

Reply:
Sorted